### PR TITLE
feat: Update signInWithRedirect to work with RN

### PR DESCRIPTION
**Pull request by cshfang**
#### Description of changes

This PR updates the `signInWithRedirect` API to use `@aws-amplify/rtn-web-browser` to open the web login in either an ephemeral `ASWebAuthenticationSession` or Custom Tab to ensure a higher security bar

#### Description of how you validated changes
* Tested with an React Native app on both iOS and Android
* `yarn test`

#### Checklist

- [x] PR description included
- [x] `yarn test` passes
- [x] Tests are [changed or added](https://github.com/aws-amplify/amplify-js/blob/main/CONTRIBUTING.md#steps-towards-contributions)

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: packages/auth/src/providers/cognito/apis/signInWithRedirect.ts [hunk 1]
```diff
@@ -105,8 +105,18 @@ function oauthSignIn({
 		.join('&');
 
 	// TODO(v6): use URL object instead
-	const URL = `https://${oauthConfig.domain}/oauth2/authorize?${queryString}`;
-	window.open(URL, SELF);
+	const oAuthUrl = `https://${domain}/oauth2/authorize?${queryString}`;
+	const { type, url } = (await openAuthSession(oAuthUrl, redirectSignIn)) ?? {};
```
Comment (HuiSF) [inline packages/auth/src/providers/cognito/apis/signInWithRedirect.ts:109]: I think we should make the `openAuthSession` function to have the same interface on both Web and native, so the logic here doesn't reply on one of the variant returning a `undefined`.  Wdyt?

**Review by HuiSF — DISMISSED** [review]

File: packages/auth/src/providers/cognito/apis/signInWithRedirect.ts (hunk 1 above)
Comment (cshfang) [reply packages/auth/src/providers/cognito/apis/signInWithRedirect.ts:109]: I think it is a fair point. The result type can't really happen but at least both can be Promisified even though the web version really doesn't need to be. Let me follow this up separately in a PR for web/expo support

**Review by jimblanc — DISMISSED** [review]
Looks good!

File: packages/core/__tests__/utils/globalHelpers/globalHelpers.native.test.ts [hunk 2]
```diff
@@ -31,8 +31,6 @@ describe('getGlobal (native)', () => {
 		});
 	});
 
-	afterAll;
```
Comment (HuiSF) [inline packages/core/__tests__/utils/globalHelpers/globalHelpers.native.test.ts:34]: After all 🤧

**Review by HuiSF — APPROVED** [review]

**Review by jimblanc — APPROVED** [review]

**Pull request merged**